lib/schemaUtils.js

```diff
@@ -3300,8 +3300,8 @@ module.exports = {
         return cb(null, mismatches);
       }
 
-      // don't validate variable if not present in transaction
-      if (!pathVar._varMatched) {
+      // don't validate variable if not present in transaction and URL path vars are not allowed
+      if (!pathVar._varMatched && !options.allowUrlPathVarMatching) {
```

Comment:
For reference, This is how `_varMatched` is defined https://github.com/postmanlabs/openapi-to-postman/blob/develop/lib/schemapack.js#L451